game/theater/frontline.py

```diff
@@ -161,9 +161,15 @@ def point_from_a(self, distance: float) -> Point:
                 )
             else:
                 remaining_dist -= segment.attack_distance
-        raise RuntimeError(
-            f"Could not find front line point {distance} from {self.blue_cp}"
-        )
+        if distance > 100:
+            # Failed to find front line point the intended distance away.
+            # Control points are probably closer to each other than FRONTLINE_MIN_CP_DISTANCE
+            # Recursively try again with shorter distance until succeeds.
+            return self.point_from_a(distance - 100)
```

Comment:
i do not understand why we do the recursive workaround in a function which should do its job for a specific given range. It should more likely happen outside this function by the caller. this would include an error handling around the function.
Maybe the problem is more bases on the actual calculation of the "5000"

Reply:
It has been confirmed to work by the original reporter. I agree that the problem might also be fixed by decreasing the FRONTLINE_MIN_CP_DISTANCE from 5000 to e.g. 3000 (which would correspond to the capture radius), but I'm not sure what other effect that would have.

Reply:
+1 for this being the wrong place. This feels like it's hiding a bug rather than fixing it.

I have a hunch (hard to say without seeing the full picture) that this could be solved with an intersection check rather than recursive back off too. Retry loops in the front line have been the dominant part of turn cycle time in the past. They should be avoided wherever possible.

Reply:
If you are ok with it, I can remove the recursion and decrease the FRONTLINE_MIN_CP_DISTANCE to 3000.

Reply:
Doesn't that just make the bug less likely?